Source/WebCore/loader/ImageLoader.cpp

```diff
@@ -486,18 +487,32 @@ void ImageLoader::notifyFinished(CachedResource& resource, const NetworkLoadMetr
         return;
     }
 
-    if (hasPendingDecodePromises())
-        decode();
-    loadEventSender().dispatchEventSoon(*this, eventNames().loadEvent);
+    Ref doc = document();
+    setShouldDelayLoadEvent(doc, true);
```

Comment:
If the document is already loaded, and we change the src value of an image element, could there be an issue with incrementing/decrementing the counter and triggering another `Document::loadEventDelayTimerFired` call?

Reply:
Nothing happens:
```
void DocumentLoader::checkLoadComplete()
{
    if (!m_frame || isLoading())
        return;
```
Earlier versions of this patch would have tested as well as they incremented and decremented for all loads and I didn't encounter issues with that.